client/src/app/pages/reports/reports.tsx

```diff
@@ -66,6 +81,34 @@ export const Reports: React.FC = () => {
     );
   }
 
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      aria-label="kebab dropdown toggle"
+      onClick={() => {
+        setIsQuestionnaireSelectOpen(!isQuestionnaireSelectOpen);
+      }}
+      isExpanded={isQuestionnaireSelectOpen}
+    >
+      {selectedQuestionnaire}
+    </MenuToggle>
+  );
+
+  const onSelectQuestionnaire = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | number | undefined
+  ) => {
+    setSelectedQuestionnaire(value as string);
+    setIsQuestionnaireSelectOpen(false);
+  };
+
+  const questionnaires = assessments.reduce((result: string[], item) => {
```

Comment:
Shouldn't need to do this IMO. The data you need is all available from the assessment object itself I think.

Reply:
@ibolton336, we need to reduce the assessments list because the same questionnaire can be used in several assessment and in order to not have repetition of the questionnaires in the select.

Reply:
We don't care about questionnaires at all. There could be 200 assessments all driven from the same questionnaire. We just care about the assessment data. I think you can safely remove every reference to questionnaires in this entire file.

Reply:
@ibolton336,

With following data set (extracted from latest) which comes directly from the assessments we need to reduce the redundant questionnaire.

```ts
[
  {
    "id": 1,
    "application": {
      "id": 1,
      "name": "App1"
    },
    "questionnaire": {
      "id": 1,
      "name": "Test questionnaire"
    },
   ..../....
 },
  {
    "id": 2,
    "application": {
      "id": 2,
      "name": "App2"
    },
    "questionnaire": {
      "id": 2,
      "name": "Test questionnaire 2"
    },
   ..../....
 },
{
    "id": 3,
    "application": {
      "id": 3,
      "name": "App3"
    },
    "questionnaire": {
      "id": 1,
      "name": "Test questionnaire"
    },
   ..../....
 },
 ```
 
Please try by yourself by creating 3 applications in which 2 of them use the same questionnaire.

Reply:
Ah Ok.